### ISSUE TITLE
Documentation Enhancement

### ISSUE THREAD

**Issue by ckoe-bccms** [SciML/OrdinaryDiffEq.jl#426]
Hello everybody,

I would like to file a "wishlist" item regarding documentation.

The original citation for an algorithm is often contained in the source file with the tableau and for new algorithms usually the github issues contain the relevant citations.

However, the user documentation, for example [http://devdocs.juliadiffeq.org/latest/internals/tableaus.html](http://devdocs.juliadiffeq.org/latest/internals/tableaus.html) or [http://docs.juliadiffeq.org/latest/solvers/ode_solve.html](http://docs.juliadiffeq.org/latest/solvers/ode_solve.html) does not contain the citations. At leats I could not find them.

It would be great if at some point the relevant citations could be made readily available directly in the documentation pages. Usually these citations contain a discussion of the problems and merits of the algorith presented, especially with respect to previously published algorithms. Easy and seamless (at least as far as possible with usually paywalled publications) would be most useful for the interested user.

Best Regards

Christof

**Comment by ChrisRackauckas** [comment]
We have a page that the front page of the documentation points to: http://juliadiffeq.org/citing.html

**Comment by ChrisRackauckas** [comment]
I like having a master page, but putting it with the solvers could be useful as well. If we got the full Zotero group actually filled out and built a bibliography it would be easier to consolidate and list these different ways.

**Comment by ckoe-bccms** [comment]
Hello everybody.

Yes, [http://juliadiffeq.org/citing.htm](http://juliadiffeq.org/citing.html) contains these references. I see this now.

Unfortunately, it is a bit hidden so that I did not find it easily. It is linked from [http://docs.juliadiffeq.org/latest/index.html](http://docs.juliadiffeq.org/latest/index.html)
with the text "... we would be grateful if you could cite our work. Please see our citation page for guidelines." For me, as a non-native english speaker at least, this looks like a link to the proper articles to cite the julia implementation, which are actually right on top of   [http://juliadiffeq.org/citing.htm](http://juliadiffeq.org/citing.html). I would not have expected to find the references to the original articles concerning the algorithms on that page, and I suspect the casual user might not scroll down far enough. Perhaps making the algorithms references on [http://juliadiffeq.org/citing.htm](http://juliadiffeq.org/citing.html) more obvious/ visible would be an easy thing to do ? Move them to their own page and link to that in the paragraph concerning citing ?

Also, [http://juliadiffeq.org/citing.htm](http://juliadiffeq.org/citing.html) alrady contains a bullett list including the DOIs. So eventually it would be easy to just link from for example a algorithm item in a list like [http://docs.juliadiffeq.org/latest/solvers/ode_solve.html#Runge-Kutta-Methods-for-Non-Stiff-Equations-1](http://docs.juliadiffeq.org/latest/solvers/ode_solve.html#Runge-Kutta-Methods-for-Non-Stiff-Equations-1) to the corresponding bullett point  on [http://juliadiffeq.org/citing.htm](http://juliadiffeq.org/citing.html) ? I am not sure if this is easy to do technically, but that is basically what I was wishing for.

Something unrelated, may be I should open a new issue although it is documentation related.  The documentation  [ http://docs.juliadiffeq.org/latest/basics/common_solver_opts.html#Stepsize-Control-1](http://docs.juliadiffeq.org/latest/basics/common_solver_opts.html#Stepsize-Control-1) says "adaptive: Turns on adaptive timestepping for appropriate methods. Default is true." . But "appropriate" (in the sense the julia implementation allows to toggle this/ uses adaptive by default I assume) is never indicated.
 
Would it be possible to document in the ODE  documentation for which algorithms adaptive step-size control is implemented  and for which ones it is possible from the implementation side  to switch that off without technical problems  ? Note: I am not talking about whether that makes mathematical sense for a particular algorithm, that is something the user has to decide after reading the original article on the solver and is nothing the julia documentation could reasonably provide. 

So, basically I am not sure what is implemented in a particular case.
Background: For a chaotic system adaptive step-size control can introduce additional noise, so it might make sense to switch that off to check the solution.

Best Regards

Christof

**Comment by ckoe-bccms** [comment]
Addendum: Or just make the heading "Algorithm Citations" on http://juliadiffeq.org/citing.htm linkable and link to that heading in the paragraph of the documentation front page explicitely ? That would avoid splitting the list of citations over multiple pages I believe.

Thank you again for your consideration !

**Comment by ChrisRackauckas** [comment]
>Also, http://juliadiffeq.org/citing.htm alrady contains a bullett list including the DOIs. So eventually it would be easy to just link from for example a algorithm item in a list like http://docs.juliadiffeq.org/latest/solvers/ode_solve.html#Runge-Kutta-Methods-for-Non-Stiff-Equations-1 to the corresponding bullett point on http://juliadiffeq.org/citing.htm ? I am not sure if this is easy to do technically, but that is basically what I was wishing for.

Yeah, that would be nice to have. It's all about getting it done. We are falling a bit behind on adding citation information: https://github.com/JuliaDiffEq/DifferentialEquations.jl/issues/273 I think we need a better way to have a live bibliography and tag the associated algorithms, and then have it generate citations on the doc pages and generate a citation doc page. I think the way to do it would be to create a big formal bibliography via Zotero where is where that issue was going, but it's all about manpower to get it done.

>Something unrelated, may be I should open a new issue although it is documentation related. The documentation http://docs.juliadiffeq.org/latest/basics/common_solver_opts.html#Stepsize-Control-1 says "adaptive: Turns on adaptive timestepping for appropriate methods. Default is true." . But "appropriate" (in the sense the julia implementation allows to toggle this/ uses adaptive by default I assume) is never indicated.

Appropriate is hard to define. It's the methods which we can actually make fixed time steps given the implementation. That's true for anything in OrdinaryDiffEq.jl, StochasticDiffEq.jl, DelayDiffEq.jl, etc. But for example you cannot do that with Sundials.jl, LSODA.jl, etc. most wrapped methods. It's more package based instead of algorithm based (since every algorithm can be fixed time step, it's whether that's exposed). We should actually word that a little differently too: Allows turning on and off adaptive timestepping (since it's normally on, so you usually use that to turn it off!). We should probably add a row to this chart:

http://docs.juliadiffeq.org/latest/basics/compatibility_chart.html

>Background: For a chaotic system adaptive step-size control can introduce additional noise, so it might make sense to switch that off to check the solution.

Well, it doesn't actually introduce noise. It's just that the sensitivity to the initial condition means that after a given amount of time the numerical solution will have diverged from the true solution, but you still get something useful because of the Shadowing Theorem. You'd probably find this cool:

http://docs.juliadiffeq.org/latest/analysis/uncertainty_quantification.html

**Comment by ckoe-bccms** [comment]
Hello,

yes, I understand. This is clearly wishlist. 

Unfortunately I have no idea how to best combine the different platforms or how much effort that means. The last time I did serious web pages was when you still wrote a web page in a text browser starting with <html> and CSS was new :-)

Thank you for the pointer to the uncertainty quantification, this looks interesting in any case.

Best Regards

Christof

**Comment by ChrisRackauckas** [comment]
The algorithms all have the references now.
